ocaml/xapi-idl/network/network_interface.ml

```diff
@@ -34,19 +34,24 @@ let comp f g x = f (g x)
 let ( ++ ) f g x = comp f g x
 
 let netmask_to_prefixlen netmask =
-  Scanf.sscanf netmask "%d.%d.%d.%d" (fun a b c d ->
-      let rec length l x = if x > 0 then length (succ l) (x lsr 1) else l in
-      let masks = List.map (( - ) 255) [a; b; c; d] in
-      32 - List.fold_left length 0 masks
+  let raise_on_ipaddr_err = function
+    | `Msg str ->
+        raise (Scanf.Scan_failure str)
```

Comment:
We are no longer using `Scanf` for parsing - so I think we should not use this exception (I understand the argument for maintaining previous behaviour). I doubt this exception is handled except at the highest level (might want to grep for it). So with could be a `failwith` which also would be turned into an internal error at the outermost level.

Reply:
Replaced with a `failwith`